### ISSUE TITLE
Default gradient clipping with max_norm=0.1

### ISSUE THREAD

**Issue by atiorh** [facebookresearch/detr#159]
Great repository and thanks for all the hard work! I have been experimenting with the panoptic network, i.e. training it from scratch using the ResNet50 backbone. During those experiments, I have observed that relaxing gradient clipping (i.e. setting the max norm to 10. instead of 0.1) significantly improves convergence. I was wondering whether the default of 0.1 is still suggested or whether it was a remnant from your experimentation phase where you were building an intuition of the training dynamics for DETR.

**Comment by alcinos** [comment]
Hi @atiorh,
Thanks for your interest in DETR.

For regular DETR training (without panoptic), gradient clipping helps with stability, and we have no evidence that relaxing it improves the convergence. Note however that clipping the norm of each layer individually (instead of the full net as implemented currently) may improve a bit the final performance (+0.2 AP IIRC)

For panoptic, we have not experimented so much with training from scratch, so I can't really tell what is going on, especially since I don't know if your other parameters are the same (batch-size, lr, ...).  You could try either removing the clipping for the panoptic head only, or do layer-wise clipping to see what you get.

Best of luck.
